_release/index.md

```diff
@@ -26,115 +26,117 @@ limitations under the License.
 
 Navigate to the release page for downloads and the changelog.
 
-* [20.0.0 (27 April 2025)][1]
-* [19.0.1 (16 February 2025)][2]
-* [19.0.0 (16 January 2025)][3]
+* [21.0.0 (17 July 2025)][1]
+* [20.0.0 (27 April 2025)][2]
+* [19.0.1 (16 February 2025)][3]
+* [19.0.0 (16 January 2025)][4]
 * [18.1.0 (24 November 2024)
-18.1.0 (2024-11-24)][4]
-* [18.0.0 (28 October 2024)][5]
-* [17.0.0 (16 July 2024)][6]
-* [16.1.0 (14 May 2024)][7]
-* [16.0.0 (20 April 2024)][8]
-* [15.0.2 (18 March 2024)][9]
-* [15.0.1 (7 March 2024)][10]
-* [15.0.0 (21 January 2024)][11]
-* [14.0.2 (19 December 2023)][12]
-* [14.0.1 (10 November 2023)][13]
-* [14.0.0 (1 November 2023)][14]
-* [13.0.0 (23 August 2023)][15]
-* [12.0.1 (13 June 2023)][16]
-* [12.0.0 (2 May 2023)][17]
-* [11.0.0 (26 January 2023)][18]
-* [10.0.1 (22 November 2022)][19]
-* [10.0.0 (26 October 2022)][20]
-* [9.0.0 (3 August 2022)][21]
-* [8.0.0 (6 May 2022)][22]
-* [7.0.0 (3 February 2022)][23]
-* [6.0.1 (18 November 2021)][24]
-* [6.0.0 (26 October 2021)][25]
-* [5.0.0 (29 July 2021)][26]
-* [4.0.1 (26 May 2021)][27]
-* [4.0.0 (26 April 2021)][28]
-* [3.0.0 (26 January 2021)][29]
-* [2.0.0 (19 October 2020)][30]
-* [1.0.1 (21 August 2020)][31]
-* [1.0.0 (24 July 2020)][32]
-* [0.17.1 (18 May 2020)][33]
-* [0.17.0 (20 April 2020)][34]
-* [0.16.0 (7 February 2020)][35]
-* [0.15.1 (1 November 2019)][36]
-* [0.15.0 (5 October 2019)][37]
-* [0.14.1 (22 July 2019)][38]
-* [0.14.0 (4 July 2019)][39]
-* [0.13.0 (1 April 2019)][40]
-* [0.12.0 (20 January 2019)][41]
-* [0.11.1 (19 October 2018)][42]
-* [0.11.0 (8 October 2018)][43]
-* [0.10.0 (6 August 2018)][44]
-* [0.9.0 (21 March 2018)][45]
-* [0.8.0 (18 December 2017)][46]
-* [0.7.1 (1 October 2017)][47]
-* [0.7.0 (17 September 2017)][48]
-* [0.6.0 (14 August 2017)][49]
-* [0.5.0 (23 July 2017)][50]
-* [0.4.1 (9 June 2017)][51]
-* [0.4.0 (22 May 2017)][52]
-* [0.3.0 (5 May 2017)][53]
-* [0.2.0 (18 February 2017)][54]
-* [0.1.0 (10 October 2016)][55]
+18.1.0 (2024-11-24)][5]
```

Comment:
Something weird going on here, but it's broken on main too. See https://arrow.apache.org/release/. I'll file an issue, can you fix this manually in this PR?

Reply:
We can keep the script as-is, this was just an issue with the 18.1.0 release note that can be fixed by https://github.com/apache/arrow-site/pull/670.